soroban-env-host/src/budget.rs

```diff
@@ -558,6 +580,54 @@ impl Budget {
         })
         .unwrap(); // impossible to panic
     }
+
+    fn get_cpu_insns_remaining_as_fuel(&self) -> Result<u64, HostError> {
+        let cpu_remaining = self.get_cpu_insns_remaining();
+        let cpu_per_fuel = self
+            .0
+            .borrow()
+            .cpu_insns
+            .get_cost_model(ContractCostType::WasmInsnExec)
+            .linear_term;
+
+        if cpu_per_fuel < 0 {
+            return Err((ScErrorType::Context, ScErrorCode::InvalidInput).into());
+        }
+        let cpu_per_fuel = (cpu_per_fuel as u64).max(1);
+        Ok(cpu_remaining / cpu_per_fuel)
```

Comment:
Ah, this .. concerns me a bit. If cpu_per_fuel doesn't evenly divide cpu_remaining, you're going to truncate the difference and throw out some fuel on each transition from VM to host and back, right?

Reply:
Yes, due to rounding, the host will give a tiny bit less fuel than it has to the VM. But the fuel is not really thrown out, that rounded off amount still stays with the host. In the end the host will subtract its budget with the amount of fuel consumed by the VM (converted back to the budget unit). 
The main consequence is the VM may run out of fuel without the host budget actually going to zero. I thought of passing a `+ 1` fuel to the VM so that it can actually run out the host's budget. But because of the other problem I mentioned before, where VM doesn't ever subtracts its fuel amount to zero (it stops right before it), it means that the possibility of this scenario (VM running out of fuel but the host budget has a small amount left) will still exist. 
I'm not sure what's the perfect solution here..

Reply:
I added a comment explaining the rounding as we discussed.